### PR TITLE
[PM-9826] Remove validation from 2fa GET and mask sensitive data

### PR DESCRIPTION
**Pull request by jlf0dev**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-9826

## 📔 Objective

Since we can't re-use TOTPs anymore, updating 2fa on accounts without a MP is currently broken. This is because we're validating the TOTP multiple times in the process (once on the GET, and once on the POST).

This removes the validation from 2fa getters, since this information mostly isn't sensitive. For the information that IS sensitive (Duo Client Secret), it masks the value before returning to user.

The TOTP is then only validated when updating the 2fa information.

Note: I also removed the validation from `/get-webauthn-challenge` and `/send-email` endpoints since they would invalidate the TOTP too early.

## 📸 Screenshots

See Jira for video

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Review by ike-kottlowski — DISMISSED** [review]
I modified the bypass to check closer to the verification in the `UserService` to only affect users using the email OTP. This keeps the experience the same for master password users.

**Comment by trmartin4** [comment]
I defaulted the `VerifySecretAsync()` additional parameter to `false` so that we kept existing logic everywhere except the new calls from the `TwoFactorController`.  This was identified with a failing test on `VerifySecretAsync_Works`

**Review by quexten — CHANGES_REQUESTED** [review]

File: src/Core/Services/IUserService.cs
[the remark this answers is not in the document]
Comment (jlf0dev) [reply src/Core/Services/IUserService.cs:1]: Done!

**Review by quexten — APPROVED** [review]
Nice work!

**Pull request merged**